### PR TITLE
Parameterize the key separator, defaulting to /

### PR DESCRIPTION
**Pull request by erikogan**
As discussed in #27, I believe there may be some utility in allowing users to specify a key separator other than /, especially when serializing to file formats other than PO/T. 

I felt it probably made the most sense to specify this configuration on the job-level, but I could easily be convinced that each plugin should have its own configuration for this value.

I’m also happy to change the names of either configuration option, I’m not wedded to either.

38 files sounds like a lot, but most of them are new tests to verify the behavior.

### CODE REVIEW

File: t/data/engine/parse_json/01/translate.serge [hunk 1]
```diff
@@ -0,0 +1,20 @@
+jobs
+{
+    {
+        @inherit                                 ../../common.serge#job_template
+
+        source_match                             \.json$
+        key_separator                            .
+        key_leading_separator                    NO
```
Comment (iafan) [inline t/data/engine/parse_json/01/translate.serge]: What is this parameter for?

**Comment by iafan** [comment]
@erikogan, please rebase this on top of latest master, since both parse_json and parse_yaml plugins have  changed

File: t/data/engine/parse_json/01/translate.serge (hunk 1 above)
Comment (erikogan) [inline t/data/engine/parse_json/01/translate.serge]: Whether or not a key should start with a leading separator: `/part/item` = YES `part.item` = NO.

The fact that you need to ask suggests the name should be changed.

**Comment by erikogan** [comment]
Oops, I thought I had. I must not have fetched before rebasing.

**Comment by erikogan** [comment]
All set. Do you have a suggestion for a better name for `key_leading_separator`?

**Comment by iafan** [comment]
Before diving into discussing implementation details, I must admit I'm not convinced this is the right approach. See https://github.com/evernote/serge/issues/27#issuecomment-242580812 where I tried to provide my rationale. Right now it looks like a fix for a non-existent problem. Provided the serializer works correctly, the only feature this change brings is the ability to fine-tune how the _comment_ is rendered to the translators. And translators won't care — any separator symbol is good. `/` was chosen as a delimiter because it is more semantically close to the notion of "path delimiter", and there's no strong reason to make it configurable.

The problem with not being able to localize files or with certain symbols in some of the original keys is the problem in the serializer, not the parser. So I'd put this PR aside for now and focus on your serializer. I'll be happy to discuss the challenges you had with it, its output format and whether this format is suited as a translation interchange format as Serge understands it.

**Comment by iafan** [comment]
Closing due to no feedback.

**Pull request closed without merging**